qiskit_experiments/framework/experiment_data.py

```diff
@@ -24,21 +24,17 @@
 class ExperimentData(DbExperimentDataV1):
     """Qiskit Experiments Data container class"""
 
-    def __init__(
-        self,
-        experiment=None,
-        backend=None,
-        job_ids=None,
-    ):
+    def __init__(self, experiment=None, backend=None, job_ids=None, parent_id=None, root_id=None):
```

Comment:
If adding these ids is necessary here, they should probably be added to `DbExperimentDataV1` instead because they will be necessary for all experiments.

Also maybe we shouldn't add them as init kwargs, but as setter methods like we do with share_level etc.

ie add to `DbExperimentdataV1` as:

```python
def __init(...):
    ...
    self._parent_id = None

@property
def parent_id(self)
    return self._parent_id

@parent_id.setter
def parent_id(self, value)
    self._parent_id = value
```

Reply:
The root is the oldest ancestor of an experiment.

As far as I know, the parent id and root id are required only for setting the tags of composite experiments and their descendants (in light of #410, I may be missing something here and there's more than I'm aware of). The parent id is in fact not necessary, so I've removed it. As for the root id, it must be passed in the constructor: https://github.com/Qiskit/qiskit-experiments/blob/c84fdf017a68dde29249d5f9d9f2881c93870f8e/qiskit_experiments/framework/composite/composite_experiment_data.py#L46-L54
Therefore, it's required as a constructor argument (and not by getters and setters) in `CompositeExperimentData` and `ExperimentData` (it's actually not used in `ExperimentData`). It's not needed in `DbExperimentDataV1`.